### PR TITLE
[Test/loc] Fix test failures with non-English language

### PR DESCRIPTION
**Pull request by wschae**
Roslyn.ExpressionEvaluator.CSharp.ResultProvider.UnitTests.dll (#1851)
Roslyn.ExpressionEvaluator.CSharp.ExpressionCompiler.UnitTests.dll (#1854)
Roslyn.ExpressionEvaluator.VisualBasic.ExpressionCompiler.UnitTests.dll (#1855)

### CODE REVIEW

**Comment by wschae** [comment]
@cston @KevinH-MS @amcasey Can you have a look at this one?

**Comment by KevinH-MS** [comment]
What are the compiler error tests doing?  I'm not completely comfortable handling these cases as one-offs (some places we do strings, some places we use the .Verify method).  There are a lot of places where we have hard-coded english strings (I'm surprised there weren't more places this had to be handled).  If we really want to use the .Verify method, we should update the test helpers to return a DiagnosticBag instead of a string.  Perhaps a better approach--can we just do "EnsureEnglishUICulture" in the ExpressionCompilerTestBase constructor (and Dispose it in the Dispose for that class)?

**Comment by wschae** [comment]
Thanks for suggestion. I will take "EnsureEnglishUICulture" approach.

**Comment by wschae** [comment]
@AlekseyTs

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/MissingAssemblyTests.cs [hunk 1]
```diff
@@ -609,7 +612,11 @@ void M()
                 out errorMessage); 
  
             Assert.Equal(2, numRetries); // Ensure that we actually retried and that we bailed out on the second retry if the same identity was seen in the diagnostics.
-            Assert.Equal($"error CS0012: The type 'MissingType' is defined in an assembly that is not referenced. You must add a reference to assembly '{missingIdentity}'.", errorMessage);
```
Comment (amcasey) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/MissingAssemblyTests.cs]: Why was this changed?  We were checking the error string because that's what we return from the public API.

Comment (wschae) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/MissingAssemblyTests.cs]: This caused the unittest failure on  non-English machines, so I wanted to avoid string comparison but I will revert the change to leave it as it is. Instead, I will take "EnsureEnglishUICulture" approach. Thanks!

Comment (amcasey) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/MissingAssemblyTests.cs]: Thanks!  You're going to do the same for the other EE tests?

File: src/ExpressionEvaluator/Core/Test/ExpressionCompiler/ExpressionCompilerTestHelpers.cs [hunk 2]
```diff
@@ -128,8 +128,10 @@ internal static class ExpressionCompilerTestHelpers
             ExpressionCompiler.CreateContextDelegate createContext,
             DkmUtilities.GetMetadataBytesPtrFunction getMetaDataBytesPtr,
             out string errorMessage,
+            out DiagnosticBag diagnosticBag,
```
Comment (cston) [inline src/ExpressionEvaluator/Core/Test/ExpressionCompiler/ExpressionCompilerTestHelpers.cs]: Does this method still need the 'errorMessage' parameter if it now exposes all diagnostics?

**Comment by cston** [comment]
LGTM, thanks.

**Comment by wschae** [comment]
We agreed to take another approach: return error code instead of string. I will close this PR.
